### PR TITLE
[HOTFIX]fixed loading issue for legacy store

### PR DESCRIPTION
**Pull request by akashrn5**
fixed loading issue for legacy store
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2831/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11091/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3062/

File: core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java [hunk 1]
```diff
@@ -84,19 +84,20 @@ private void addDimensions(List<CarbonDimension> dimensions) {
...
-    short actualPosition = 0;
+    short dictActualPosition = 0;
+    short noDimActualPosition = 0;
     // sort step's output is based on sort column order i.e sort columns data will be present
     // ahead of non sort columns, so table spec also need to add dimension spec in same manner
     for (int i = 0; i < dimensions.size(); i++) {
       CarbonDimension dimension = dimensions.get(i);
       if (dimension.isComplex()) {
-        spec = new DimensionSpec(ColumnType.COMPLEX, dimension, actualPosition++);
+        spec = new DimensionSpec(ColumnType.COMPLEX, dimension, (short) 0);
```
Comment (qiuchenjian) [inline core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java]: Why is it '0' ?

File: core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java [hunk 2]
```diff
@@ -84,19 +84,20 @@ private void addDimensions(List<CarbonDimension> dimensions) {
     List<DimensionSpec> dictDimensionSpec = new ArrayList<>();
     int dimIndex = 0;
     DimensionSpec spec;
-    short actualPosition = 0;
+    short dictActualPosition = 0;
+    short noDimActualPosition = 0;
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java]: Please rename as `noDictActualPosition`

File: core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java (hunk 1 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java]: Here also please keep them in `noDimActualPosition`

Comment (akashrn5) [reply core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java]: done

File: core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java (hunk 2 above)
Comment (akashrn5) [reply core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java]: done, please review and merge

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3063/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3064/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2832/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2833/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3065/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2834/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2835/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11095/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3066/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3067/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2836/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11096/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2837/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11097/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3068/

**Pull request closed without merging**